selectel/dbaas_postgresql_utils.go

```diff
@@ -0,0 +1,85 @@
+package selectel
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/selectel/dbaas-go"
+)
+
+func resourceDBaaSPostgreSQLDatastoreV1PoolerFromSet(poolerSet *schema.Set) (*dbaas.Pooler, error) {
+	if poolerSet.Len() == 0 {
+		return nil, nil
+	}
+	var resourceModeRaw, resourceSizeRaw interface{}
+	var ok bool
+
+	resourcePoolerMap := poolerSet.List()[0].(map[string]interface{})
+	if resourceModeRaw, ok = resourcePoolerMap["mode"]; !ok {
+		return &dbaas.Pooler{}, errors.New("pooler.mode value isn't provided")
+	}
+	if resourceSizeRaw, ok = resourcePoolerMap["size"]; !ok {
+		return &dbaas.Pooler{}, errors.New("pooler.size value isn't provided")
+	}
+
+	resourceMode := resourceModeRaw.(string)
+	resourceSize := resourceSizeRaw.(int)
+
+	pooler := &dbaas.Pooler{
+		Mode: resourceMode,
+		Size: resourceSize,
+	}
+
+	return pooler, nil
+}
+
+func resourceDBaaSPostgreSQLDatastoreV1PoolerOptsFromSet(poolerSet *schema.Set) (dbaas.DatastorePoolerOpts, error) {
```

Comment:
NIT: looks like `resourceDBaaSPostgreSQLDatastoreV1PoolerOptsFromSet` and `resourceDBaaSPostgreSQLDatastoreV1PoolerFromSet` do the same things but return different result and it is possible to call first func to get opts and init pooler in second func.

Reply:
Not sure that I understood this correctly. But I created separate function for poolerSet parsing to remove duplicate code.
There are two different function, because Pooler return fields and create fields potentially can change in the future